app/views/layouts/user_header.html.erb

```diff
@@ -39,24 +39,47 @@
           <% end %>
         </div>
       </div>
-    </div>
-
-    <% if user_profile_page? %>
-      <div class="col-6 user-header__info--margin-0 align-self-center d-lg-none">
-        <h1 class="user-header__name">
-          <%= @user.name %>
-        </h1>
-        <p class="user-header__location #{show_location?}">
-          <%= @user.location %>
-        </p>
-      </div>
-      <% if user_profile_page? %>
-        <div class="col-6 user-header__nav-wrapper">
-          <div class="edit-profile-button-container edit-profile-button-container--d-lg-none d-flex justify-content-end">
-            <%= link_to "Edit Profile", edit_user_path(@user), class: "btn btn-white align-self-end flex-row-reverse" %>
+      <% if user_signed_in? && current_user == @user %>
```

Comment:
There seems to be more than one place where you check for if user is signed in and it's the current user. We can maybe reduce this down to one helper method (probably can put current_user==@user inside user_signed_in?)

Also it's good practice to create template HTML files to reuse some of the repeated HTML markups (kind of like react components if you are familiar with them)